tests/language-feature/capability/var-capability-wgsl-2.slang

```diff
@@ -0,0 +1,23 @@
+//TEST:SIMPLE(filecheck=CHECK): -target wgsl -entry Main -stage compute
+//TEST:SIMPLE(filecheck=CHECK_IGNORE_CAPS): -target wgsl -entry Main -stage compute -ignore-capabilities
+
+//CHECK:error 36108: 'InputTyped' has dependencies that are not compatible on the required target 'wgsl'.
+//CHECK_IGNORE_CAPS-NOT: error 36108
+struct inputWrapper {
+    [require(cpp_cuda_glsl_hlsl_metal_spirv)]
```

Comment:
I don't think we should be adding `[require]` to parameter/var declarations. Instead they should be added to the `struct RWBuffer` decl, and we should infer the requirement of a vardeclbase from its type.